lib/awaitableSender.ts

```diff
@@ -170,14 +170,19 @@ export class AwaitableSender extends BaseSender {
    * @param {number} [format] The message format. Specify this if a message with custom format needs
    * to be sent. `0` implies the standard AMQP 1.0 defined format. If no value is provided, then the
    * given message is assumed to be of type Message interface and encoded appropriately.
+   * @param {number} [timeoutInSeconds] If provided, this timeout overrides the `sendTimeoutInSeconds` 
+   * that is set when the `AwaitableSender` is created. This timeout represents the duration in which 
+   * the promise to send the message should complete (resolve/reject). If not, the Promise will be 
+   * rejected after timeout.
    * @returns {Promise<Delivery>} Promise<Delivery> The delivery information about the sent message.
    */
-  send(msg: Message | Buffer, tag?: Buffer | string, format?: number): Promise<Delivery> {
+  send(msg: Message | Buffer, tag?: Buffer | string, format?: number, timeoutInSeconds?: number): Promise<Delivery> {
     return new Promise<Delivery>((resolve, reject) => {
       log.sender("[%s] Sender '%s' on amqp session '%s', credit: %d available: %d",
         this.connection.id, this.name, this.session.id, this.credit,
         this.session.outgoing.available());
       if (this.sendable()) {
+        if (timeoutInSeconds && typeof timeoutInSeconds == "number" && timeoutInSeconds > 0) this.sendTimeoutInSeconds = timeoutInSeconds;
```

Comment:
I would flip this around to something like
```suggestion
        let sendTimoutInSeconds = this.sendTimeoutInSeconds;
        if (typeof timeoutInSeconds === "number" && timeoutInSeconds > 0) sendTimeoutInSeconds = timeoutInSeconds;
```
And then update the `setTimeout` call on the next line to use `sendTimeoutInSeconds` instead of `this.sendTimeoutInSeconds`. I don't think the timeoutInSeconds passed to send should ever replace the existing `this.sendTimeoutInSeconds` value, but rather be used instead if it is available.